akka-http/src/main/scala/akka/http/javadsl/server/EntityStreamingSupport.scala

```diff
@@ -0,0 +1,144 @@
+   * however really are just plain JSON or something else internally (perhaps with slight extensions).
+   *
+   * NOTE: Implementations should specialize the return type to their own Type!
+   */
+  def withContentType(range: ContentType): EntityStreamingSupport
+
+  /**
+   * Write-side / read-side, defines if (un)marshalling should be done in parallel.
+   *
+   * This may be beneficial marshalling the bottleneck in the pipeline.
+   *
+   * See also [[parallelism]] and [[withParallelMarshalling]].
+   */
+  def parallelism: Int
+
+  /**
+   * Write-side / read-side, defines if (un)marshalling of incoming stream elements should be perserved or not.
+   *
+   * Allowing for parallel and unordered (un)marshalling often yields higher throughput and also allows avoiding
+   * head-of-line blocking if some elements are much larger than others.
+   *
+   * See also [[parallelism]] and [[withParallelMarshalling]].
+   */
+  def unordered: Boolean
+
+  /**
+   * Write-side / read-side, defines parallelism and if ordering should be preserved or not of Source element marshalling.
+   *
+   * Sometimes marshalling multiple elements at once (esp. when elements are not evenly sized, and ordering is not enforced)
+   * may yield in higher throughput.
+   *
+   * NOTE: Implementations should specialize the return type to their own Type!
+   */
+  def withParallelMarshalling(parallelism: Int, unordered: Boolean): EntityStreamingSupport
+
+}
+
+/**
+ * Entity streaming support, independent of used Json parsing library etc.
+ */
+object EntityStreamingSupport {
+
+  /**
+   * Default `application/json` entity streaming support.
+   *
+   * Provides framing (based on scanning the incoming dataBytes for valid JSON objects, so for example uploads using arrays or
+   * new-line separated JSON objects are all parsed correctly) and rendering of `\n` separated JSON objects for rendering Sources
+   * as streams of new-line separated JSON objects.
+   *
+   * Limits the maximum JSON object length to 8KB, if you want to increase this limit provide a value explicitly.
+   *
+   * See also <a href="https://en.wikipedia.org/wiki/JSON_Streaming">https://en.wikipedia.org/wiki/JSON_Streaming</a>
+   */
+  def json(): JsonEntityStreamingSupport = json(8 * 1024)
+  /**
+   * Default `application/json` entity streaming support.
+   *
+   * Provides framing (based on scanning the incoming dataBytes for valid JSON objects, so for example uploads using arrays or
+   * new-line separated JSON objects are all parsed correctly) and rendering of `\n` separated JSON objects for rendering Sources
+   * as streams of new-line separated JSON objects.
+   *
+   * See also <a href="https://en.wikipedia.org/wiki/JSON_Streaming">https://en.wikipedia.org/wiki/JSON_Streaming</a>
+   */
+  def json(maxObjectLength: Int): JsonEntityStreamingSupport = akka.http.scaladsl.server.EntityStreamingSupport.json(maxObjectLength)
```

Comment:
Maybe later if people demand it? Adding methods to EntityStreamingSupport is a huge pain, as it forces everyone to support that